datafusion/core/tests/sqllogictests/README.md

```diff
@@ -105,10 +105,13 @@ query <type_string> <sort_mode>
 
 - `test_name`: Uniquely identify the test name (arrow-datafusion only)
 - `type_string`: A short string that specifies the number of result columns and the expected datatype of each result column. There is one character in the <type_string> for each result column. The characters codes are:
-  - "T" for a text result,
-  - "I" for an integer result,
-  - "R" for a floating-point result,
-  - "?" for any other type.
+  - 'B' - **B**oolean,
+  - 'D' - **D**atetime,
+  - 'I' - **I**nteger,
+  - 'F' - **F**loating-point results,
+  - 'S' - **S**string,
+  - 'T' - **T**imestamp,
+  - "?" - any other types
```

Comment:
Honestly, I wanted to add only `Boolean` at first but then reshuffled the letters to accommodate more types.

Notable differences here:
- `F` for the floating points. Sqllite defines floating points as `R`. [Cocroach](https://github.com/cockroachdb/cockroach/blob/899c9112188c22ec140adbeb0cf7d5761da3827d/pkg/sql/logictest/logic.go#L285-L287) differentiates floating points `F`, and decimals `R`
- `S` for strings. Sqllite and Cocroach define these as `T` text. But I repurposed `T` for **t**imestamps.

I tried using first letters of the words that have some meaning, but I am open for suggestions here.

Reply:
I think we should stick as close as possibly to other standards -- so therefore I suggest:

 `T` for text
 `P` for timestamP

I also think it would be good to differentiate between `F` and `R` if possible 

What do you think?

Reply:
> I think we should stick as close as possibly to other standards -- so therefore I suggest:
> T for text
> P for timestamP

Sounds good. It would probably make it easier to use tests suites from other databases, if you ever want to run them.
Updated to be https://github.com/apache/arrow-datafusion/pull/5253/commits/7e32de81b1c238cd5932de707f12f8f40a0c9c9b.

> I also think it would be good to differentiate between F and R if possible

This one is trickier. It would be hard to check Postgres/Datafusion compatibility on floating points then. Postgres usually falls back to using `Numeric` by default, which is their decimal. Whereas Datafusion prefers Floats unless otherwise specified.

I could technically provide a custom validator for postgres that treats expected results the same way for Postgres. So the Postgres' validator would treat `query FR` the same way as `query RR`. 
If you don't think having different validators is confusing, I can do it.

Reply:
> This one is trickier. It

If it is tricky I don't think it is needed then